storybookjs/storybook
# Addon-docs: Fix embedding selected story in canvas block

## What I did

Fixed an issue where embedding the currently selected story in a canvas block resulted in a preview crash.

Previously the `CURRENT_SELECTION` token was resolved to a real story ID only if the Source component was passed as a single `id` prop, but since the Canvas component passes an `ids` array to Source regardless of how many children are present, the handling was skipped.

To fix, the Source component was modified to handle `CURRENT_SELECTION` if it appears anywhere in `id` or `ids`. This occurs before IDs are passed to `useStories` since the story store expects resolved story IDs instead of the `"."` token.

## How to test

Open the `Addons/Docs/mdx` story and verify that the new example for a selected story in a canvas renders correctly with source.
